### PR TITLE
DateBox: valueChanged event should be fired after input clearing undo (T878918)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by DokaRus — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 1]
```diff
@@ -1029,6 +1029,28 @@ module('Options changed', setupModule, () => {
         assert.strictEqual(this.$input.val(), 'October 10 2012', 'date is not changed on mouse wheel');
     });
 
+    test('ValueChanged event should be fired after input clearing undo (T878918)', function(assert) {
+        const optionChangedHandler = sinon.spy();
+        const date = this.instance.option('value');
+        this.instance.option('onValueChanged', optionChangedHandler);
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]:
```suggestion
        this.instance.option('onValueChanged', valueChangedHandler);
```

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js [hunk 2]
```diff
@@ -1029,6 +1029,28 @@ module('Options changed', setupModule, () => {
...
+        assert.ok(optionChangedHandler.calledOnce, 'value has been changed');
+        assert.strictEqual(args.value, null, 'value is correct');
+
+        this.$input
+            .val(date)
+            .change();
+
+        args = optionChangedHandler.getCall(1).args[0];
+        assert.strictEqual(optionChangedHandler.callCount, 2, 'value has been changed');
+        assert.strictEqual(args.value.toString(), date.toString(), 'value is correct');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]:
```suggestion
        assert.deepEqual(args.value, date, 'value is correct');
```

Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js (hunk 1 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Review by San4es — COMMENTED** [review]
LGTM

**Pull request merged**